tests/components/nest/test_climate_sdm.py

```diff
@@ -1259,7 +1259,8 @@ async def test_thermostat_missing_temperature_trait(
     assert ATTR_FAN_MODE not in thermostat.attributes
     assert ATTR_FAN_MODES not in thermostat.attributes
 
-    await common.async_set_temperature(hass, temperature=24.0)
+    with pytest.raises(KeyError):
```

Comment:
We should fail with a friendlier error message, for example look at how the preset mode check is handled when a mode is not supported.

Reply:
Rather: With an explicit exception, rather than a runtime exception. (Alternatively If we think this is a "shouldn't happen" case, then don't bother with a test.)

Reply:
I believe this to be a "it should never happen" branch.  That said, since it is explicitly coded and explicitly tested perhaps it does occur.  I did have a situation where setpoints completely stopped working with NEST; and I suppose it is possible it was silently exiting via this branch.

Either way, I've changed it to throw a specific HomeAssistantError and the error includes the list of traits that is in the device_info.